osquery/filesystem/filesystem.cpp

```diff
@@ -139,20 +140,23 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
     boost::filesystem::directory_iterator begin_iter(path);
     boost::filesystem::directory_iterator end_iter;
     for (; begin_iter != end_iter; begin_iter++) {
-      if (!boost::filesystem::is_directory(begin_iter->path())) {
-        results.push_back(begin_iter->path().string());
+      try {
+        if (boost::filesystem::is_regular_file(begin_iter->path())) {
+          results.push_back(begin_iter->path().string());
+        }
+      } catch (const boost::filesystem::filesystem_error& e) {
+        if (ignore_error == 0) {
```

Comment:
Is there ever a callsite where you don't want to ignore filesystem errors? They're common and unpredictable, can we default-expect our API to return with FS-error resilience?